### DISCUSSION TITLE
FastAPI doc display schema of pydantic generic model with wrong name

### DISCUSSION THREAD

**Discussion by IterableTrucks** [fastapi/fastapi#9893] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Generic, Optional, TypeVar

from fastapi import FastAPI
from pydantic import BaseModel
from pydantic.generics import GenericModel

DataT=TypeVar('DataT')
class GenericResponse(GenericModel,Generic[DataT]):
    data:Optional[DataT]
class Profile(BaseModel):
    name:str

api=FastAPI()

@api.get("/")
def generic_model()->GenericResponse[Profile]:
    return GenericResponse(data=Profile(name='Author'))
```

### Description

Schema's name is not GenericResponse[Profile] but GenericResponse_Profile_ instead.(pydantic version is 1.10.11). The OpenAPI json generated by fastapi is:
```json
{
  "openapi": "3.1.0",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "get": {
        "summary": "Generic Model",
        "operationId": "generic_model__get",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/GenericResponse_Profile_"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "GenericResponse_Profile_": {
        "properties": {
          "data": {
            "$ref": "#/components/schemas/Profile"
          }
        },
        "type": "object",
        "title": "GenericResponse[Profile]"
      },
      "Profile": {
        "properties": {
          "name": {
            "type": "string",
            "title": "Name"
          }
        },
        "type": "object",
        "required": [
          "name"
        ],
        "title": "Profile"
      }
    }
  }
}
```
Fastapi replaces the '[' ']' characters with underscores and relies on `title` field to display the schema's name. But the Swagger-UI 5.1 has dropped supporting `title`  field, which causes this problem.

### Operating System

Linux

### FastAPI Version

0.99.1

### Python Version

3.11.2

**Comment by jgould22** [comment, marked as answer]
For context 

https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.1.md#components-object

> All the fixed fields declared above are objects that MUST use keys that match the regular expression: ^[a-zA-Z0-9\.\-_]+$.

So that is why the `[` `]` are removed.

However it seems to work in https://editor-next.swagger.io/ so 🤷🏻

**Comment by steve148** [comment]
~Coming in late to this conversation, but does it make sense to remove the trailing slash on the right side of the schema name? So instead of `GenericResponse_Profile_` we would have `GenericResponse_Profile`. If this warrants a new discussion post, let me know.~

Further investigation, it looks like the name for this case is being generated by pydantic, so not really a fastapi thing. Ignore me.
